File: erlang/accumulate/accumulate.erl
-module(accumulate).
-export([accumulate/2]).

accumulate(Fn, List)       -> accumulate(Fn,List,[]).
accumulate(_,  [],    Acc) -> Acc;
accumulate(Fn, [H|T], Acc) -> [Fn(H)|accumulate(Fn, T, Acc)].

File: clojure/accumulate/src/accumulate.clj
(ns accumulate
  "The Accumulate exercise from Exercism")

(defn accumulate
  "Return a lazy sequence consisting of the result of applying `f` to
  each item in `coll`."
  [f coll]
  (lazy-seq
   (when-let [[x & xs] (seq coll)]
     (cons (f x) (accumulate f xs)))))
